src/Entities/EntityEffect.cpp

```diff
@@ -373,7 +373,7 @@ void cEntityEffectRegeneration::OnTick(cPawn & a_Target)
 	}
 
 	// Regen frequency = 50 ticks, divided by potion level (Regen II = 25 ticks)
-	int frequency = FloorC(50.0 / static_cast<double>(m_Intensity + 1));
+	int frequency = FloorC(50.0 / std::min(static_cast<double>(m_Intensity + 1), 50.0));
```

Comment:
Nit: IMO the division by 0 is more obvious written this way:
```suggestion
	int frequency = std::max(1, FloorC(50.0 / static_cast<double>(m_Intensity + 1));
```